### DISCUSSION TITLE
Child class "extend" constructor

### DISCUSSION THREAD

**Discussion by TahirAhmadov** [dotnet/csharplang#5601] [Language Ideas]
Can we get something like this?
```cs
class Base
{
  public Base(int x, int y) { ... }
}
class Child : Base
{
  public extend Child(int z) : base() { ... }
  // this actually becomes
  public Child(int x, int y, int z) : base(x, y) { ... }
}
var c = new Child(1, 2, 3);
```
I'm not sure what syntax to use here; I imagine this can be done with some new keyword, or using an existing contextual keyword, or some symbol. Basically, if the base class only has one "visible" (not `private`) constructor, we allow a child class to "extend" this constructor - and thus saving a lot of typing and maintenance.

**Comment by huoyaoyuan** [comment]
See #806

**Reply by TahirAhmadov** [reply]
Thanks. That discussion talks about only propagating base ctors; I added a post for extending them, as well.

**Reply by huoyaoyuan** [reply]
> extending them

Propagating is likely to happen. Extending will be very likely rejected.
In C# we care a lot about API clarity. The caller will want to see all the parameters they need to pass.

**Reply by TahirAhmadov** [reply]
The caller will see the parameters. That child constructor is identical to listing out all the parameters from the base constructor first, and passing them along to the base. Furthermore, in the IDE, I envision that mousing over the child constructor definition will show all the parameters, too, as a helpful visual reminder.

**Reply by huoyaoyuan** [reply]
> Furthermore, in the IDE

Although we have first-class IDE support, readability without IDE is still considered a lot.

**Reply by TahirAhmadov** [reply]
> Although we have first-class IDE support, readability without IDE is still considered a lot.

Yes but that's also part of the benefit here: I'm talking about complicated constructors accepting many parameters with lengthy type names and parameter names. Trying to look at code with these multi-line constructor invocations is often more difficult, not less so, than just seeing the `extend` keyword (or whatever). This typical initialization code (base constructor parameters being passed to the base) doesn't really matter much in the context of code review, for example.

**Reply by huoyaoyuan** [reply]
Another problem is which constructor to extend when the base class has multiple.
Extending with a keyword is somehow acceptable then,

**Reply by TahirAhmadov** [reply]
> Another problem is which constructor to extend when the base class has multiple. Extending with a keyword is somehow acceptable then,

I specifically clarified that in the OP: "if the base class only has one "visible" (not private) constructor,"
